spark/src/main/scala/geotrellis/spark/io/s3/S3LayerDeleter.scala

```diff
@@ -0,0 +1,55 @@
+package geotrellis.spark.io.s3
+
+import geotrellis.spark.{Boundable, KeyBounds, LayerId}
+import geotrellis.spark.io.index.KeyIndex
+import geotrellis.spark.io._
+import org.apache.spark.SparkContext
+import spray.json.JsonFormat
+import geotrellis.spark.io.json._
+import spray.json.DefaultJsonProtocol._
+import scala.reflect.ClassTag
+
+class S3LayerDeleter[K: JsonFormat: ClassTag]
+  (attributeStore: AttributeStore[JsonFormat])(implicit sc: SparkContext) extends LayerDeleter[LayerId] {
+
+  def getS3Client: () => S3Client = () => S3Client.default
+
+  // TODO: mb to add here numPartitions: Int arg? (trick as in reader)
+  def delete(id: LayerId): Unit = {
+    if (!attributeStore.layerExists(id)) throw new LayerNotFoundError(id)
+    val (header, _, keyBounds, keyIndex, _) = try {
+      attributeStore.readLayerAttributes[S3LayerHeader, Unit, KeyBounds[K], KeyIndex[K], Unit](id)
+    } catch {
+      case e: AttributeNotFoundError => throw new LayerDeleteError(id).initCause(e)
+    }
+
+    val bucket = header.bucket
+    val prefix = header.key
+    val numPartitions = sc.defaultParallelism
```

Comment:
:+1: i wonder, will it be better just to get list of keys for the `header.key`, then to iterate over them, group by 1024 keys per request?
This will help us to avoid index decomposing and to get rid of type param.